burstphoto/texture/texture.swift

```diff
@@ -78,10 +79,16 @@ func add_texture_highlights(_ in_texture: MTLTexture, _ out_texture: MTLTexture,
 }
 
 
-func add_texture_exposure(_ in_texture: MTLTexture, _ out_texture: MTLTexture, _ norm_texture: MTLTexture, _ exposure_bias: Int, _ white_level: Int, _ black_level: [Int]) {
+func add_texture_exposure(_ in_texture: MTLTexture, _ out_texture: MTLTexture, _ norm_texture: MTLTexture, _ exposure_bias: Int, _ white_level: Int, _ black_level: [Int], _ color_factors: [Double]) {
     
     let black_level_mean = 0.25*Double(black_level[0] + black_level[1] + black_level[2] + black_level[3])
     
+    let color_factor_mean = 0.25*(color_factors[0]+2.0*color_factors[1]+color_factors[2])
+    
+    let in_texture_blurred = blur(in_texture, with_pattern_width: 1, using_kernel_size: 1)
+    
+    let weight_highlights_texture_blurred = calculate_weight_highlights(in_texture, exposure_bias, white_level, black_level_mean)
```

Comment:
I think you need a description of why you're creating a blurred texture here

Reply:
Thanks, I added a description.